deployment/Triton/README.md

```diff
@@ -0,0 +1,277 @@
+```
+$ python -u client/client.py [filename/directory]
+```
+and the program returns
+``` Default input for the client is client/test_data/prostate_24.nii.gz
+$ Classification result: ['Prostate segmented']
+```
+## Examples:
+The example demonstrates running a Triton Python Backend on a single image classification problem.  
+1. First, a Dockerfile and build script is used to build a container to Run the Triton Service and copy the model specific files in the container. 
+```Dockerfile:
+# use desired Triton container as base image for our app
+FROM nvcr.io/nvidia/tritonserver:21.04-py3
+
+# create model directory in container
+RUN mkdir -p /models/monai_covid/1
+
+# install project-specific dependencies
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+RUN rm requirements.txt
+
+# copy contents of model project into model repo in container image
+COPY models/monai_covid/config.pbtxt /models/monai_covid
+COPY models/monai_covid/1/model.py /models/monai_covid/1
+COPY models/monai_covid/1/covid19_model.ts /models/monai_covid/1
+
+ENTRYPOINT [ "tritonserver", "--model-repository=/models"]
+#ENTRYPOINT [ "tritonserver", "--model-repository=/models", "--log-verbose=1"]
+```
+Note: The Triton service expects a certain directory structure discussed in [Model Config File](#model-config-file) to load the model definitions.
+
+2. Next, the container with the Triton Service is run as a service (in background or separate terminal for demo).  In this example, the ports used by the Triton Service are set to `8090` for client communications.
+```Dockerfile:
+demo_app_image_name="monai_triton:demo"
+docker run --shm-size=128G --rm -p 127.0.0.1:8090:8000 -p 127.0.0.1:8091:8001 -p 127.0.0.1:8092:8002 ${demo_app_image_name}
+```
+3. See [Model Config File](#model-config-file) to see the expected file structure for Triton.
+- Modify the models/monai_prostrate/1/model.py file to satisfy any model configuration requirements while keeping the required components in the model definition. See the * [Usage](#usage) section for background. 
+- In the models/monai_prostrate/1/config.pbtxt file configure the number of GPUs and which ones are used.
+e.g. Using two available GPUs and two parallel versions of the model per GPU
+```
+instance_group [
+  { 
+    kind: KIND_GPU
+    count: 2
+    gpus: [ 0, 1 ]
+  }
+```
+e.g. Using three of four available GPUs and four parallel versions of the model per GPU
+```
+instance_group [
+  { 
+    kind: KIND_GPU
+    count: 4
+    gpus: [ 0, 1, 3 ]
+  }
+```
+Also, other configurations like dynamic batching and corresponding sizes can be configured. See the [Triton Service Documentation model configurations](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md) documentation for more information.
+ 
+- Finally, be sure to include you Tensors or Torchscript definition *.ts files in the directory structure. In this example,a prostate segmentation model based in PyTorch is used. 
```

Comment:
typo in `include you`?

Thanks.

Reply:
- Finally, be sure to include Tensors or Torchscript definition *.ts files in the directory structure. In this example, a COVID19 classificatiion model based in PyTorch is used. 
```
covid19_model.ts
```
The Dockerfile will copy the model definition structure into the Triton container Service.  When the container is run, the python backend implementation will pull the covid19_model.ts file from a Google Drive for the demo.  So the container should be rebuilt after any modifications to the GPU configuration or model configurations for the example.